src/domain/pricing/editModal/EditForm.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { FunctionComponent } from 'react';
+import { Form, Formik } from 'formik';
+import { Button } from 'hds-react/lib';
+import { useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
+
+import styles from './editForm.module.scss';
+import {
+  AdditionalService,
+  BerthPrice,
+  HarborService,
+  WinterStoragePrice,
+} from '../PricingPage';
+import EditFormFields, { PRICING_TYPES } from './fields/EditFormFields';
+import { getAdditionalServicesValidationSchema } from './fields/AdditionalServicesFields';
+import { getBerthsValidationSchema } from './fields/BerthsFields';
+import { getWinterStorageValidationSchema } from './fields/WinterStorageFields';
+import { getHarborServicesValidationSchema } from './fields/HarborServicesFields';
+
+export interface EditPricingFormProps {
+  onSubmit: (
+    values: BerthPrice | WinterStoragePrice | HarborService | AdditionalService
+  ) => void;
+  closeModal: () => void;
+  initialValues:
+    | BerthPrice
+    | WinterStoragePrice
+    | HarborService
+    | AdditionalService;
+  formType: PRICING_TYPES;
+}
+
+const getEditFormValidationSchema = (formType: PRICING_TYPES, t: TFunction) => {
+  switch (formType) {
+    case PRICING_TYPES.BERTHS:
+      return getBerthsValidationSchema(t);
+    case PRICING_TYPES.WINTER_STORAGE:
+      return getWinterStorageValidationSchema(t);
+    case PRICING_TYPES.HARBOR_SERVICES:
+      return getHarborServicesValidationSchema(t);
+    case PRICING_TYPES.ADDITIONAL_SERVICES:
+      return getAdditionalServicesValidationSchema(t);
+  }
```

Comment:
Having the same switch statement, here and in `EditFormFields` makes the code a little more verbose.
Not sure whether it would be better to encapsulate them in the same module and import them from there, or even make completely four separate components. since it doesn't seem that we're gaining many benefits from sharing the same `EditForm` component. 

What if `EditModal` takes whatever `children` we pass to instead of just `EditForm`? Then we can use it like:
```TSX 
<EditModal {...props}>
   <BerthsFields />
</EditModal>
```
And get rid of `EditFormFields` module entirely.

Reply:
I kept EditModal intact, but merged EditFormFields with EditForm

Reply:
Alright, looks better :)